clang/lib/CodeGen/TargetInfo.cpp

```diff
@@ -9970,6 +9979,34 @@ unsigned SPIRTargetCodeGenInfo::getOpenCLKernelCallingConv() const {
   return llvm::CallingConv::SPIR_KERNEL;
 }
 
+bool SPIRTargetCodeGenInfo::shouldEmitStaticExternCAliases() const {
+  return false;
+}
+
+LangAS SPIRTargetCodeGenInfo::getGlobalVarAddressSpace(CodeGenModule &CGM,
+                                                       const VarDecl *D) const {
+  assert(!CGM.getLangOpts().OpenCL &&
+         !(CGM.getLangOpts().CUDA && CGM.getLangOpts().CUDAIsDevice) &&
+         "Address space agnostic languages only");
+  LangAS DefaultGlobalAS = getLangASFromTargetAS(
+      CGM.getContext().getTargetAddressSpace(LangAS::opencl_global));
+  if (!D)
+    return DefaultGlobalAS;
+
+  LangAS AddrSpace = D->getType().getAddressSpace();
+  assert(AddrSpace == LangAS::Default || isTargetAddressSpace(AddrSpace) ||
+         // allow applying clang AST address spaces in SYCL mode
+         (CGM.getLangOpts().SYCL && CGM.getLangOpts().SYCLIsDevice));
+  if (AddrSpace != LangAS::Default)
+    return AddrSpace;
+
+  if (CGM.isTypeConstant(D->getType(), false)) {
+    if (auto ConstAS = CGM.getTarget().getConstantAddressSpace())
```

Comment:
Can you spell out `auto`?

Reply:
This is a copy-paste from [AMDGPUTargetCodeGenInfo::getGlobalVarAddressSpace](https://github.com/intel/llvm/blob/sycl/clang/lib/CodeGen/TargetInfo.cpp#L9089-L9091). 

All the places where `getConstantAddressSpace` is called, authors use `auto`: [CodeGenModule::getStringLiteralAddressSpace](https://github.com/llvm/llvm-project/blob/bb9eb198298099742c823dce11c5edacc9c48d4e/clang/lib/CodeGen/CodeGenModule.cpp#L4053), [castStringLiteralToDefaultAddressSpace](https://github.com/llvm/llvm-project/blob/bb9eb198298099742c823dce11c5edacc9c48d4e/clang/lib/CodeGen/CodeGenModule.cpp#L4071) and [createReferenceTemporary](https://github.com/llvm/llvm-project/blob/55488bd3cd1a468941e26ad4cf94f2bad887fc02/clang/lib/CodeGen/CGExpr.cpp#L405).

I'm okay to apply this comment, but I think its should be applied to all instances. Right?
If so, I can create an NFC patch and merge it before - https://reviews.llvm.org/D89909. Does it sound okay to you?

Reply:
Your call -- I called it out because the type wasn't obvious from the context and on line 9996 we're doing `LangAS AddrSpace = D->getType().getAddressSpace();` with the type spelled out, so I figured we might as well spell it out. However, given how trivial the use is on the next line, it's not critical.

Reply:
I'll ask John McCall (CodeGen code owner) in https://reviews.llvm.org/D89909 about this.